File: src-tauri/src/models/system_skill_tree_node.rs
use serde::Serialize;

use crate::models::SystemSkill;

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct SystemSkillTreeNode {
    pub id: String,
    pub name: String,
    pub path: String,
    pub kind: String,
    pub skill: Option<SystemSkill>,
    pub children: Vec<SystemSkillTreeNode>,
}

File: src/types/system-skill-tree-node.ts
import type { SystemSkill } from "./system-skill";
import type { SystemSkillTreeFile } from "./system-skill-tree-file";
import type { SystemSkillTreeNodeKind } from "./system-skill-tree-node-kind";

export type SystemSkillTreeNode = {
  id: string;
  name: string;
  path: string;
  kind: SystemSkillTreeNodeKind;
  skill: SystemSkill | null;
  file: SystemSkillTreeFile | null;
  children: SystemSkillTreeNode[];
};
